RHDEVDOCS-6022: Content creation for namespace-scoped Argo Rollouts installation (#78933)
openshift/openshift-docs: modules/gitops-configuring-namespace-scoped-argo-rollouts-installation.adoc

// Module included in the following assemblies:
//
// * argo_rollouts/enable-support-for-namespace-scoped-argo-rollouts-installation.adoc

:_mod-docs-content-type: PROCEDURE
[id="configuring-namespace-scoped-argo-rollouts-installation_{context}"]

> = Configuring namespace-scoped Argo Rollouts installation

Reviewer: ```suggestion
= Configuring a namespace-scoped Argo Rollouts installation
```
Instead of saying "Configuring a...installation", would it be accurate to simply say "Installing a namespace-scoped Argo Rollouts instance"?

Author: Fixed.

> To install a namespace-scoped instance of Argo Rollouts, complete the following steps:

Reviewer 2: Shouldn't end in a colon, and module title should match the intro/shortdesc. See https://redhat-documentation.github.io/modular-docs/#con-creating-procedure-modules_writing-mod-docs.

Reviewer 2: Also, am I configuring an installation or installing something?

Author: Fixed.

> The following example provides the steps to configure namespace-scoped Argo Rollouts installation.

Reviewer: ```suggestion
The following example provides the steps to configure a namespace-scoped Argo Rollouts installation.
```

Reviewer: Consider revising this line. Is the purpose of this task to show an example, or is it to provide the steps to install a namespace-scoped instance of Argo Rollouts? What goal is the customer trying to reach by reading this content?

Author: The intention is to provide the steps to install a namespace-scoped instance of Argo Rollouts. Changing the line to as follows: To install a namespace-scoped instance of Argo Rollouts, complete the following steps:

> To configure a namespace-scoped instance of Argo Rollouts installation, complete the following steps:

Reviewer 3: Should really end in a period here. Not enough to hold up a merge review, but just calling it out. 

```suggestion
To configure a namespace-scoped instance of Argo Rollouts installation, complete the following steps.
```

Author: Fixed.

.Prerequisites

> * You have logged in to the {OCP} cluster as an administrator.

Reviewer: ```suggestion
* You have logged in to the {product-title} cluster as an administrator.
```

Author: Fixed.

> * You have logged in to the {product-title} cluster as an administrator.

Reviewer 2: ```suggestion
* You are logged in to the {product-title} cluster as an administrator.
```
I think it's reasonable to use present tense. You want to the reader to definitely be logged in as they start this. Just a suggestion.

Author: Fixed.

> * You have installed {gitops-title} on your {OCP} cluster.

Reviewer: ```suggestion
* You have installed {gitops-title} on your {product-title} cluster.
```

Author: Fixed.

.Procedure 

. In the *Administrator* perspective of the web console, go to *Administration* -> *CustomResourceDefinitions*.

> . Search for `Subscription` and click the `Subscription` CRD.

Reviewer: ```suggestion
. Search for `Subscription` and click the *Subscription* CRD.
```

Author: Fixed.

> . Click the *Instances* tab and then click the `openshift-gitops-operator` subscription.

Reviewer: ```suggestion
. Click the *Instances* tab and then click the *openshift-gitops-operator* subscription.
```

Author: Fixed.

. Click the *YAML* tab and edit the YAML file. 
+

> Specify the `NAMESPACE_SCOPED_ARGO_ROLLOUTS` environment variable, with the value set to `true` in the `.spec.config.env` property.

Reviewer: ```suggestion
.. Specify the `NAMESPACE_SCOPED_ARGO_ROLLOUTS` environment variable, with the value set to `true` in the `.spec.config.env` property.
```

Author: Fixed.

+

> .Example of configuring the namespace-scoped Argo Rollouts installation
> +
> [source,yaml]

Reviewer 4: ```suggestion
.Example of configuring the namespace-scoped Argo Rollouts installation
[source,yaml]
```

Author: Fixed.

apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: openshift-gitops-operator
spec:
  # (...)
  config:
    env:
      - name: NAMESPACE_SCOPED_ARGO_ROLLOUTS

>         value: 'true' #<1>

Reviewer 5: ```suggestion
        value: true #<1>
```

Please check with the SME once and clarify.

----

> <1> Set value to `'true'` to enable namespace-scoped installation. If the value is set to `'false'` or not specified, the installation defaults to cluster-scoped mode.

Reviewer 5: ```suggestion
<1> The value set to `true` enables namespace-scoped installation. If the value is set to `'false'` or not specified the installation defaults to cluster-scoped mode.
```

About the quotation marks around the value true and false, please check with the SME and clarify whether to include the quotation marks or not.

Author: Thanks for the detailed explanation, @jgwest. Fixed.

> . Click *Save*.

Reviewer: ```suggestion
.. Click *Save*.
```

Author: Fixed.

+

> As a result, the {gitops-title} Operator enables the namespace-scoped Argo Rollouts installation to reconcile Argo Rollouts CR.

Reviewer: ```suggestion
The {gitops-title} Operator enables the namespace-scoped Argo Rollouts installation to reconcile the Argo Rollouts CR.
```

Author: Fixed.

> The {gitops-title} Operator enables the namespace-scoped Argo Rollouts installation to reconcile the Argo Rollouts CR.

Reviewer 2: ISG prefers some alternative to "enables" unless you're talking about activating something, essentially. Consider rewriting to avoid this issue.

Author: Fixed.

> . Verify if the {gitops-title} Operator has enabled namespace-scoped Argo Rollouts installation by viewing the logs of the {gitops-shortname} container:

Reviewer: ```suggestion
. Verify that the {gitops-title} Operator has enabled the namespace-scoped Argo Rollouts installation by viewing the logs of the {gitops-shortname} container:
```

Author: Fixed.

.. In the *Administrator* perspective of the web console, go to *Workloads* -> *Pods*.

> .. Click the `openshift-gitops-operator-controller-manager` pod, and then click the *Logs* tab.

Reviewer: ```suggestion
.. Click the *openshift-gitops-operator-controller-manager* pod, and then click the *Logs* tab.
```

Author: Fixed.

> .. Look for the following log statement: `Running in namespaced-scoped mode`. This indicates that the {gitops-title} Operator has enabled the namespace-scoped Argo Rollouts installation.

Reviewer: ```suggestion
.. Look for the following log statement: `Running in namespaced-scoped mode`. This statement indicates that the {gitops-title} Operator has enabled the namespace-scoped Argo Rollouts installation.
```

Author: Fixed.

> . Create a `RolloutManager` resource to perform the namespace-scoped Argo Rollout installation:

Reviewer: ```suggestion
. Create a `RolloutManager` resource to complete the namespace-scoped Argo Rollout installation:
```

Author: Fixed.

> . Create a `RolloutManager` resource to complete the namespace-scoped Argo Rollout installation:

Reviewer 5: ```suggestion
. Create a `RolloutManager` resource to complete the namespace-scoped Argo Rollouts installation:
```

Author: Fixed.

.. Go to *Operators* -> *Installed Operators* -> *Red Hat OpenShift GitOps*, and click the *RolloutManager* tab.

> .. Click the *Create RolloutManager* button.

Reviewer: ```suggestion
.. Click *Create RolloutManager*.
```

Author: Fixed.

.. Select *YAML view* and enter the following snippet:
+

> .Example RolloutManager CR for namespace-scoped Argo Rollouts installation

Reviewer: ```suggestion
.Example `RolloutManager` CR for a namespace-scoped Argo Rollouts installation
```

Author: Fixed.

.Example RolloutManager CR for namespace-scoped Argo Rollouts installation
[source,yaml]
----
apiVersion: argoproj.io/v1alpha1
kind: RolloutManager
metadata:
  name: rollout-manager
  namespace: my-application #<1> 
spec:
  namespaceScoped: true
----

> <1> This indicates the name of the project to perform the namespace-scoped Argo Rollout installation

Reviewer 4: it should be more action-centric:
```suggestion
<1> Indicate the name of the project where you want to perform the namespace-scoped Argo Rollout installation.
```

Author: Fixed.

> <1> Indicate the name of the project where you want to perform the namespace-scoped Argo Rollout installation.

Reviewer: ```suggestion
<1> Indicate the name of the project where you want to install the namespace-scoped Argo Rollout instance.
```

Author: Fixed.

> <1> Indicate the name of the project where you want to install the namespace-scoped Argo Rollout instance.

Reviewer 5: ```suggestion
<1> Specify the name of the project where you want to install the namespace-scoped Argo Rollouts instance.
```

Author: Fixed.

> <1> This indicates the name of the project to perform the namespace-scoped Argo Rollout installation
> .. Click *Create*. 

Reviewer 4: Add space between the two lines:
```suggestion
<1> This indicates the name of the project to perform the namespace-scoped Argo Rollout installation

.. Click *Create*. 
```

Author: Fixed.

+

> After the `RolloutManager` CR is created, the {gitops-title} begins the namespace-scoped Argo Rollouts installation into the selected namespace.

Reviewer: ```suggestion
After the `RolloutManager` CR is created, {gitops-title} begins to install the namespace-scoped Argo Rollouts instance into the selected namespace.
```

Author: Fixed.

> . Verify the successful namespace-scoped installation.  

Reviewer: ```suggestion
. Verify that the namespace-scoped installation was successful.
```

Author: Fixed.

> . Verify that the namespace-scoped installation was successful.

Reviewer 5: ```suggestion
. Verify that the namespace-scoped installation is successful.
```

> . Verify that the namespace-scoped installation is successful.

Reviewer 2: This could be the start of a verification section. 

https://redhat-documentation.github.io/modular-docs/#verification

Author: Agreed but as the verification section is optional I do not see any major benefit of carving out an explicit verification section. I think the content looks fine as is now. WDYT @maxwelldb?

.. In the *RolloutManager* tab, under the *RolloutManagers* section, ensure that the *Status* field of the `RolloutManager` instance is `Phase: Available`.
.. Examine the following output in the *YAML* tab under the *RolloutManagers* section to ensure that the installation is successful:
+
.Example of namespace-scoped Argo Rollouts installation YAML file
[source,yaml]
----
spec:
  namespaceScoped: true
status:
  conditions:
    lastTransitionTime: '2024-07-10T14:20:5z`
    message: ''
    reason: Success
    status: 'True' #<1>
    type: 'Reconciled'
  phase: Available
  rolloutController: Available
----

> <1> This indicates that the namespace-scoped Argo Rollouts installation is enabled successfully.

Reviewer: ```suggestion
<1> This status indicates that the namespace-scoped Argo Rollouts installation is enabled successfully.
```

Author: Fixed.

+

> You can see an error message if a namespace-specific Argo Rollouts installation is attempted while a cluster-scoped installation already exists on the cluster:

Reviewer: ```suggestion
If you try to install a namespace-specific Argo Rollouts instance while a cluster-scoped installation already exists on the cluster, an error message is displayed:
```

Author: Fixed.

+
.Example of an incorrect installation with an error message
[source,yaml]
----
spec:

>   namespaceScoped: false

Reviewer 6: ```suggestion
  namespaceScoped: true
```

Author: Fixed.

status:
  conditions:
   lastTransitionTime: '2024-07-10T14:10:7z`

>    message: 'when Subscription has environment variable Namespace_Scoped_Argo_RolloutsManagers: In this case, only namespace-scoped RolloutManager resources are supported'

Reviewer 7: You can change the `message:` line to `message: when Subscription has environment variable NAMESPACE_SCOPED_ARGO_ROLLOUTS set to False, there may not exist any namespace-scoped RolloutManagers: only a single cluster-scoped RolloutManager is supported`

Here is the line in the code where that message is coming from, for reference: https://github.com/argoproj-labs/argo-rollouts-manager/blob/192dd2c3b5dd026e2c59c5765e98ca2f70ca01f9/controllers/utils.go#L22C46-L22C248

Author: Fixed.

   reason: InvalidRolloutManagerScope
   status: 'False' #<1>
   type: 'Reconciled'
  phase: Failure
  rolloutController: Failure
----

> <1> This indicates that the namespace-scoped Argo Rollouts installation is not enabled successfully. The installation defaults to cluster-scoped mode.

Reviewer: ```suggestion
<1> This status indicates that the namespace-scoped Argo Rollouts installation is not enabled successfully. The installation defaults to cluster-scoped mode.
```

Author: Fixed.

// Module included in the following assemblies:
//
// * argo_rollouts/enable-support-for-namespace-scoped-argo-rollouts-installation.adoc

:_mod-docs-content-type: PROCEDURE
[id="configuring-namespace-scoped-argo-rollouts-installation_{context}"]
= Configuring namespace-scoped Argo Rollouts installation

The following example provides the steps to configure namespace-scoped Argo Rollouts installation.

.Prerequisites

* You have logged in to the {OCP} cluster as an administrator.
* You have installed {gitops-title} on your {OCP} cluster.

.Procedure 

. In the *Administrator* perspective of the web console, go to *Administration* -> *CustomResourceDefinitions*.
. Search for `Subscription` and click the `Subscription` CRD.
. Click the *Instances* tab and then click the `openshift-gitops-operator` subscription.
. Click the *YAML* tab and edit the YAML file. 

> . Specify the `NAMESPACE_SCOPED_ARGO_ROLLOUTS` environment variable, with the value set to `true` in the `.spec.config.env` property.

Reviewer 4: I would add this as part of previous step, as it is still the same step: editing the yaml file:
```suggestion
+
Specify the `NAMESPACE_SCOPED_ARGO_ROLLOUTS` environment variable, with the value set to `true` in the `.spec.config.env` property.
```

Author: Fixed.

+
.Example of configuring the namespace-scoped Argo Rollouts installation
+
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: openshift-gitops-operator
spec:
  # (...)
  config:
    env:
      - name: NAMESPACE_SCOPED_ARGO_ROLLOUTS
        value: 'true' #<1>
----

> <1> Set value to `true` to enable namespace-scoped installation. If the value is set to `false` or not specified, the installation defaults to cluster-scoped mode.

Reviewer 4: Because the ' ' around true and false are neccessarry, I would include it in the example value:
```suggestion
<1> Set value to `'true'` to enable namespace-scoped installation. If the value is set to `'false'` or not specified, the installation defaults to cluster-scoped mode.
```

Author: Fixed.

. Click *Save*.
+
As a result, the {gitops-title} Operator enables the namespace-scoped Argo Rollouts installation to reconcile Argo Rollouts CR.

. Verify if the {gitops-title} Operator has enabled namespace-scoped Argo Rollouts installation by viewing the logs of the {gitops-shortname} container:
.. In the *Administrator* perspective of the web console, go to *Workloads* -> *Pods*.
.. Click the `openshift-gitops-operator-controller-manager` pod, and then click the *Logs* tab.
.. Look for the following log statement: `Running in namespaced-scoped mode`. This indicates that the {gitops-title} Operator has enabled the namespace-scoped Argo Rollouts installation.

. Create a `RolloutManager` resource to perform the namespace-scoped Argo Rollout installation:

> .. Create or select the project where you want to perform the namespace-scoped Argo Rollout installation from the *Project* drop-down menu.

Reviewer 4: This step can be ommited if you add the callout in the YAML for namespace, as we discussed :+1: 
```suggestion

```

Author: Fixed.

.. Go to *Operators* -> *Installed Operators* -> *Red Hat OpenShift GitOps*, and click the *RolloutManager* tab.
.. Click the *Create RolloutManager* button.
.. Select *YAML view* and enter the following snippet:
+

> .Example RolloutManager CR for namespace-scoped Argo Rollouts installation
> +
> [source,yaml]
> ----
> apiVersion: argoproj.io/v1alpha1

Reviewer 4: Remove `+` between title and the snippet:
```suggestion
.Example RolloutManager CR for namespace-scoped Argo Rollouts installation
[source,yaml]
----
apiVersion: argoproj.io/v1alpha1
```

Author: Fixed.

kind: RolloutManager
metadata:
  name: rollout-manager
spec:
  namespaceScoped: true

> ----

Reviewer 4: ```suggestion
----
<1> Specify the namespace where you want to perform the namespace-scoped Argo Rollout installation.
```

Reviewer 4: @Dhruv-Soni11 One more minor suggestion here regarding this: https://github.com/openshift/openshift-docs/pull/78933#discussion_r1692697916

Reviewer 4: to make the namespace generic

Author: Fixed.

.. Click *Create*. 
+
After the `RolloutManager` CR is created, the {gitops-title} begins the namespace-scoped Argo Rollouts installation into the selected namespace.

. Verify the successful namespace-scoped installation.  
.. In the *RolloutManager* tab, under the *RolloutManagers* section, ensure that the *Status* field of the `RolloutManager` instance is `Phase: Available`.
.. Examine the following output in the *YAML* tab under the *RolloutManagers* section to ensure that the installation is successful:
+

> .Example of namespace-scoped Argo Rollouts installation YAML file
> +
> [source,yaml]
> ----

Reviewer 4: ```suggestion
.Example of namespace-scoped Argo Rollouts installation YAML file
[source,yaml]
----
```

Author: Fixed.

. Verify if the {gitops-title} Operator has enabled namespace-scoped Argo Rollouts installation by viewing the logs of the {gitops-shortname} container:
.. In the *Administrator* perspective of the web console, go to *Workloads* -> *Pods*.
.. Click the `openshift-gitops-operator-controller-manager` pod, and then click the *Logs* tab.
.. Look for the following log statement: `Running in namespaced-scoped mode`. This indicates that the {gitops-title} Operator has enabled the namespace-scoped Argo Rollouts installation.

. Create a `RolloutManager` resource to perform the namespace-scoped Argo Rollout installation:
.. Create or select the project where you want to perform the namespace-scoped Argo Rollout installation from the *Project* drop-down menu.
.. Go to *Operators* -> *Installed Operators* -> *Red Hat OpenShift GitOps*, and click the *RolloutManager* tab.
.. Click the *Create RolloutManager* button.
.. Select *YAML view* and enter the following snippet:
+
.Example RolloutManager CR for namespace-scoped Argo Rollouts installation
+
[source,yaml]
----
apiVersion: argoproj.io/v1alpha1
kind: RolloutManager
metadata:
  name: rollout-manager

>   namespace: openshift-gitops-operator #<1> 

Reviewer 4: ```suggestion
  namespace: example-namespace #<1> 
```

Author: Fixed.

// Module included in the following assemblies:
//
// * argo_rollouts/enable-support-for-namespace-scoped-argo-rollouts-installation.adoc

:_mod-docs-content-type: PROCEDURE
[id="configuring-namespace-scoped-argo-rollouts-installation_{context}"]
= Configuring namespace-scoped Argo Rollouts installation

The following example provides the steps to configure namespace-scoped Argo Rollouts installation.

.Prerequisites

* You have logged in to the {OCP} cluster as an administrator.
* You have installed {gitops-title} on your {OCP} cluster.

> * You have installed the {gitops-title} Operator on your {OCP} cluster.

Reviewer 7: These two lines are the same.

Author: Fixed.

.Procedure 

> . In the *Administrator* perspective, navigate to *Administration* → *CustomResourceDefinitions*.

Author: Fixed.

> . Search for the `Subscription` CRD and click to open it.

Author: Fixed.

> . Select the *Instances* tab and click the `openshift-gitops-operator` subscription.

Author: Fixed.

> . Select the *YAML* tab and make your customization. 

Author: Fixed.

> . To configure the operator for enabling the namespace-scoped Argo Rollouts installation, specify the `NAMESPACE_SCOPED_ARGO_ROLLOUTS` environment variable, with the value set to 'true' (single quotes required) in the `.spec.config.env` property.

Author: Fixed.

+

> .Example `NAMESPACE_SCOPED_ARGO_ROLLOUTS` environment variable configuration

Author: Fixed.

+
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: openshift-gitops-operator
spec:
  # (...)
  config:
    env:
      - name: NAMESPACE_SCOPED_ARGO_ROLLOUTS
        value: 'true' #<1>
----

> <1> - Specifies the value of the environment variable for enabling namespace-scoped Argo Rollouts installation

Author: Fixed.

. Click *Save*.
+
As a result, the {gitops-title} Operator enables the namespace-scoped Argo Rollouts installation to reconcile Argo Rollouts CR.

> . You can also verify whether the {gitops-title} Operator has enabled namespace-scoped Argo Rollouts installation by viewing the logs of the Openshift GitOps container. 

Author: Fixed.

> .. In the *Administrator* perspective, navigate to *Workloads* → *Pods*. 

Author: Fixed.

> .. Select the `openshift-gitops-operator-controller-manager` Pod, and then select the *Logs* tab.

Author: Fixed.

+

> As a result, the container logs display a log statement "Running in namespaced-scoped mode", which indicates that the {gitops-title} Operator has enabled the namespace-scoped Argo Rollouts installation.

Author: Fixed.

> . Create a RolloutManager resource to perform the namespace-scoped Argo Rollout installation. 

Author: Fixed.

.. Create or select the project where you want to perform the namespace-scoped Argo Rollout installation from the *Project* drop-down menu.

> .. Navigate to *Operators* → *Installed Operators* → *Red Hat OpenShift GitOps*, and select the *RolloutManager* tab.

Author: Fixed.

.. Click the *Create RolloutManager* button.
.. Select *YAML view*. When creating a namespace-scoped Argo Rollouts installation, you must set the field '.spec.namespaceScoped:` to `true` in the RolloutManager CR. 
.. Enter the following snippet in the *YAML view*.

> +

Author: Fixed.

> .Example - RolloutManager CR for namespace-scoped Argo Rollouts installation

Author: Fixed.

+
[source,yaml]
----
apiVersion: argoproj.io/v1alpha1
kind: RolloutManager
metadata:

>   name: rollout-manager

Reviewer 4: ```suggestion
  name: rollout-manager
  namespace: example-namespace #<1>
```

>   name: rollout-manager

Author: Fixed.

spec:
  namespaceScoped: true
----

.. Click *Create*. 
+

> After the RolloutManager CR is created, {gitops-title} installs a namespace-scoped Argo Rollouts installation into the namespace.

Author: Fixed.

> . To verify the successful installation of namespace-scoped Argo Rollouts installation, in the *RolloutManager* tab, under the *RolloutManagers* section, verify that the *Status* field of the RolloutManager instance shows as *Phase: Available*.

Author: Fixed.

+

> You can also examine the following output in the *YAML* tab under the *RolloutManagers* section to ensure that the installation is successful.

Author: Fixed.

+

> .Example - namespace-scoped Argo Rollouts installation output in the YAML tab

Author: Fixed.

+
[source,yaml]
----
spec:

>   namespaceScoped: true

Author: Fixed.

status:
  conditions:
    lastTransitionTime: '2024-07-10T14:20:5z`
    message: ''
    reason: Success
    status: 'True'
    type: 'Reconciled'
  phase: Available
  rolloutController: Available
----

> +
> This output indicates that namespace-specific Argo Rollouts installation is enabled successfully. For instance, an error message will appear if a namespace-specific Argo Rollouts installation is attempted while a cluster-wide installation already exists on the cluster.

Reviewer 4: instead of this, it would be great to add something like the following:
```suggestion
+
You can see an error message if a namespace-specific Argo Rollouts installation is attempted while a cluster-scoped installation already exists on the cluster:
+
.Example of an incorrect installation with an error message
[source,ymal]
----
<Show the example here>
----
```

Author: Fixed.
